e2e/main.py

```diff
@@ -157,5 +161,152 @@ def call_checker_endpoint(name, param, amount=0):
+        ctez = deploy_ctez(
+            self.client,
+            ctez_dir=os.path.join(PROJECT_ROOT, "vendor/ctez"),
+            ttl=MAX_OPERATIONS_TTL,
+        )
+
+        print("Deploying Checker.")
+        checker = deploy_checker(
+            self.client,
+            checker_dir=CHECKER_DIR,
+            oracle=oracle.context.address,
+            ctez=ctez["fa12_ctez"].context.address,
+            ttl=MAX_OPERATIONS_TTL,
+        )
+
+        print("Deployment finished.")
+
+        def call_endpoint(contract, name, param, amount=0):
+            print("Calling", contract.key.public_key_hash(), "/", name, "with", param)
+            return inject(
+                self.client,
+                getattr(contract, name)(param)
+                .with_amount(amount)
+                .as_transaction()
+                .autofill(ttl=MAX_OPERATIONS_TTL)
+                .sign(),
+            )
+
+        def call_bulk(bulks, *, chunk_size):
+            chunks = [
+                bulks[i : i + chunk_size] for i in range(0, len(bulks), chunk_size)
+            ]
+            for chunk_no, chunk in enumerate(chunks, 1):
+                print(
+                    "Sending",
+                    len(bulks),
+                    "operations as bulk:",
+                    "Chunk",
+                    chunk_no,
+                    "of",
+                    len(chunks),
+                )
+                inject(
+                    self.client,
+                    self.client.bulk(*chunk).autofill(ttl=MAX_OPERATIONS_TTL).sign(),
+                )
+
+        call_bulk(
+            [
+                checker.create_burrow((0, None)).with_amount(200_000_000),
+                checker.mint_kit((0, 80_000_000)),
+            ],
+            chunk_size=10,
+        )
+
+        call_endpoint(
+            ctez["ctez"], "create", (1, None, {"any": None}), amount=2_000_000
+        )
+        call_endpoint(ctez["ctez"], "mint_or_burn", (1, 100_000))
+        call_endpoint(ctez["fa12_ctez"], "approve", (checker.context.address, 100_000))
+
+        call_endpoint(
+            checker,
+            "add_liquidity",
+            (100_000, 100_000, 5, int(datetime.now().timestamp()) + 20),
+        )
+
+        burrows = list(range(1, 1001))
+
+        call_bulk(
+            [
+                checker.create_burrow((burrow_id, None)).with_amount(100_000_000)
+                for burrow_id in burrows
+            ],
+            chunk_size=100,
+        )
+
+        # Mint as much as possible from the burrows. All should be identical, so we just query the
+        # first burrow and mint that much kit from all of them.
+        max_mintable_kit = checker.metadata.burrowMaxMintableKit(
+            (self.client.key.public_key_hash(), 1)
+        ).storage_view()
+
+        bulks = []
+        for burrow_no in burrows:
+            bulks.append(
+                checker.touch_burrow((self.client.key.public_key_hash(), burrow_no))
+            )
```

Comment:
Do we still need this call to `touch_burrow` here? Now that we touch burrows implicitly when we `mint_kit` the explicit call to touch_burrow should be unnecessary, right?
```suggestion
```

Reply:
Oh right, this test was implemented before this change. I'll remove the touch_burrow call.